hdate/zmanim.py

```diff
@@ -57,9 +68,73 @@ def zmanim(self):
         return {key: self.utc_minute_timezone(value) for
                 key, value in self.get_utc_sun_time_full().items()}
 
+    @property
+    def candle_lighting(self):
+        """Return the time for candle lighting, or None if not applicable."""
+        today_holiday_type = HDate(
+            gdate=self.date, diaspora=self.location.diaspora).holiday_type
+
```

Comment:
If instead of directly pulling the holiday type, you use `today = HDate(...)`, you can use `today.is_shabbat` and today.is_yom_tov` making the code easier to read.